PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['crm.lead']
-    _inherit = ['crm.team']
-    _inherit = ['res.users']
-    _inherit = ['sale.order']

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_crm/models/crm_lead.py
+++ b/addons/sale_crm/models/crm_lead.py
@@ -8,7 +8,7 @@ from odoo.osv import expression
 
 
 class CrmLead(models.Model):
-    _inherit = ['crm.lead']
+    _inherit = 'crm.lead'
 
     sale_amount_total = fields.Monetary(compute='_compute_sale_data', string="Sum of Orders", help="Untaxed Total of Confirmed Orders", currency_field='company_currency')
     quotation_count = fields.Integer(compute='_compute_sale_data', string="Number of Quotations")

--- a/addons/sale_crm/models/crm_team.py
+++ b/addons/sale_crm/models/crm_team.py
@@ -6,7 +6,7 @@ from odoo.tools import SQL
 
 
 class CrmTeam(models.Model):
-    _inherit = ['crm.team']
+    _inherit = 'crm.team'
 
     def _compute_dashboard_button_name(self):
         super(CrmTeam, self)._compute_dashboard_button_name()

--- a/addons/sale_crm/models/res_users.py
+++ b/addons/sale_crm/models/res_users.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     target_sales_invoiced = fields.Integer('Invoiced in Sales Orders Target')

--- a/addons/sale_crm/models/sale_order.py
+++ b/addons/sale_crm/models/sale_order.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class SaleOrder(models.Model):
-    _inherit = ['sale.order']
+    _inherit = 'sale.order'
 
     opportunity_id = fields.Many2one(
         'crm.lead', string='Opportunity', check_company=True,
